source/extensions/filters/http/lua/wrappers.cc

```diff
@@ -118,8 +124,38 @@ void HeaderMapWrapper::checkModifiable(lua_State* state) {
   }
 }
 
+int HeaderMapWrapper::luaSetHttp1ReasonPhrase(lua_State* state) {
+  checkModifiable(state);
+
+  size_t input_size = 0;
+  const char* phrase = luaL_checklstring(state, 2, &input_size);
+
+  Envoy::Http::StatefulHeaderKeyFormatterOptRef formatter(headers_.formatter());
+
+  if (!formatter.has_value()) {
+    using envoy::extensions::http::header_formatters::preserve_case::v3::
+        PreserveCaseFormatterConfig;
+    using Envoy::Http::ResponseHeaderMapImpl;
+    using Envoy::Http::StatefulHeaderKeyFormatter;
+    using Http::HeaderFormatters::PreserveCase::PreserveCaseHeaderFormatter;
+
+    ResponseHeaderMapImpl* map = dynamic_cast<ResponseHeaderMapImpl*>(&headers_);
+    if (map) {
```

Comment:
This dynamic cast is necessary because this could be called on request headers also? Can you add a comment? It would be nice to avoid the cast but probably not worth the effort to track request/response some other way.

Reply:
Yes, pretty much. Also since I'm not that familiar with Envoy code yet and I wasn't sure there could be other (current or future) `headers_` implementations.